spec/recurly/purchase_spec.rb

```diff
@@ -42,72 +43,141 @@
     )
   end
 
-  describe "Purchase.invoice!" do
-    it "should return an invoice_collection when valid" do
+  describe 'Purchase.invoice!' do
+    it 'should return an invoice_collection when valid' do
       stub_api_request(:post, 'purchases', 'purchases/invoice-201')
       collection = Purchase.invoice!(purchase)
       collection.charge_invoice.must_be_instance_of Invoice
       shipping_address = collection.charge_invoice.line_items.first.shipping_address
       shipping_address.must_be_instance_of ShippingAddress
     end
-    it "should raise an Invalid error when data is invalid" do
+
+    it 'should return an invoice_collection with ramp intervals' do
+      stub_api_request(:post, 'purchases', 'purchases/invoice-with-ramp-pricing-201')
+      collection = Purchase.invoice!(purchase)
+      charge_invoice = collection.charge_invoice
+
+      charge_invoice.total_in_cents.must_equal 7000
```

Comment:
Some of these tests aren't really testing the class (`Purchase`, in this case) as much as they are testing the underlying superclass methods (i.e. the way `Resource` deserializes responses) - not a huge deal, but we have stayed away from such tests in the previous work.

Reply:
Mudei a descrição da especificação para uma mais específica

Reply:
I Changed the spec description to a more specific